File: host/src/main/java/dk/dtu/domain/rules/effects/RebootToken.java
package dk.dtu.domain.rules.effects;

import dk.dtu.domain.core.Phase;
import dk.dtu.domain.model.Direction;
import dk.dtu.domain.model.Tile;
import dk.dtu.domain.rules.api.BoardAPI;

import java.util.EnumSet;

/**
 * Reboot token tile effect that restores destroyed robots during the reboot phase.
 * @author Weihao Mo
 */
public record RebootToken(Direction direction) implements TileEffect {

    /**
     * @author Weihao Mo
     */
    @Override
    public void onPhase(Phase phase, Tile tile, BoardAPI api) {
    }

    /**
     * @author Weihao Mo
     */
    @Override
    public EnumSet<Phase> phases() {
        return null;
    }
}

File: engine/src/effects/rebootToken.ts
import { Phase } from "../core/phase.js";
import type { Direction } from "../model/direction.js";
import type { Tile } from "../model/tile.js";
import type { BoardAPI } from "../rules/boardApi.js";
import type { TileEffect } from "./tileEffect.js";

/** Ported from dk.dtu.domain.rules.effects.RebootToken. */
export class RebootToken implements TileEffect {
  constructor(readonly direction: Direction) {}

  onPhase(_phase: Phase, _tile: Tile, _api: BoardAPI): void {
    // Respawn is driven by Game.applyRespawnPhase, not by a phase pass.
  }

  phases(): Set<Phase> | null {
    return null;
  }
}
